### PR TITLE
gcsfuse 0.11.2

### PR DESCRIPTION
**Pull request by jacobsa**
This release comes with a new build tool and with vendored dependencies,
so the extensive list of dependencies is no longer needed.

### CODE REVIEW

**Comment by jacobsa** [comment]
@dunn: To punish you for being so helpful and responsive in other pull requests recently, can I ask you to look at this one? :-) It seems like this repository isn't watched for pull requests as closely as the main one; just want to make sure it doesn't get lost.

File: Formula/gcsfuse.rb [hunk 1]
```diff
@@ -1,96 +1,34 @@
...
-  end
+  depends_on "daemon"
+  depends_on "go" => :build
 
   def install
-    ENV["GOPATH"] = buildpath
-    Language::Go.stage_deps resources, buildpath/"src"
+    # Build the build_gcsfuse tool. Ensure that it doesn't pick up any
+    # libraries from the user's GOPATH; it should have no dependencies.
+    ENV["GOPATH"] = "/not/a/real/directory"
```
Comment (DomT4) [inline Formula/gcsfuse.rb]: Use `ENV.delete["GOPATH"]` here.

File: Formula/gcsfuse.rb [hunk 2]
```diff
@@ -1,96 +1,34 @@
...
-    # Make the gcsfuse source code from the tarball appear in the appropriate
-    # place in the $GOPATH workspace, then build the binaries with names that
-    # don't conflict with directory names.
-    mkdir_p "src/github.com/googlecloudplatform"
-    ln_s buildpath, "src/github.com/googlecloudplatform/gcsfuse"
-    system "go", "build", "-o", "gcsfuse.bin"
-    system "go", "build", "-o", "gcsfuse_mount_helper.bin", "./gcsfuse_mount_helper"
+    # Use that tool to build gcsfuse itself.
+    if build.head?
+      gcsfuse_version = "#{`git rev-parse --short HEAD`.strip}"
```
Comment (DomT4) [inline Formula/gcsfuse.rb]: Does 

```ruby
= `git rev-parse --short HEAD`.strip
```
not work?

File: Formula/gcsfuse.rb [hunk 3]
```diff
@@ -1,96 +1,34 @@
...
+    else
+      gcsfuse_version = version
+    end
 
-    bin.install "gcsfuse.bin" => "gcsfuse"
-    bin.install "gcsfuse_mount_helper.bin" => "gcsfuse_mount_helper"
+    ohai "Using gcsfuse version name: #{gcsfuse_version}"
```
Comment (DomT4) [inline Formula/gcsfuse.rb]: Let's not `ohai`.

File: Formula/gcsfuse.rb (hunk 1 above)
Comment (DomT4) [inline Formula/gcsfuse.rb]: Or keep it as `ENV["GOPATH"] = buildpath` if Go complains about the lack of set `GOPATH`.

File: Formula/gcsfuse.rb (hunk 3 above)
Comment (jacobsa) [inline Formula/gcsfuse.rb]: Done.

File: Formula/gcsfuse.rb (hunk 2 above)
Comment (jacobsa) [inline Formula/gcsfuse.rb]: Thanks, it does. (I cargo culted this from another formula; my Ruby isn't great.)

File: Formula/gcsfuse.rb (hunk 1 above)
Comment (jacobsa) [inline Formula/gcsfuse.rb]: Much better, done.

**Comment by jacobsa** [comment]
Thanks for the comments, everything addressed.

**Comment by dunn** [comment]
Looks good! Merged in 036060cc3e72f10f520ef19497eaf211b98f0b0b.

**Comment by jacobsa** [comment]
@dunn: Thanks!

**Pull request closed without merging**